[maps] clarify how layer filters are applied to term joins (#128502)
elastic/kibana: docs/maps/search.asciidoc

You can apply a search request to individual layers by setting `Filters` in the layer details panel.
Click the *Add filter* button to add a filter to a layer.

> NOTE: Layer filters are not applied to the right side of *term joins*. You can apply a search request to the right side of *term joins* by setting the *where* clause in the join definition. For example, a term join where the left side is roads and the right side is traffic volumn measurements, a layer filter of `roadType is "highway"` would only be applied to roads index and not applied to the traffic volumn measurements index.

Reviewer: ```suggestion
NOTE: Layer filters are not applied to the right side of *term joins*. You can apply a search request to the right side of *term joins* by setting the *where* clause in the join definition. For example, a term join where the left side is roads and the right side is traffic volume measurements, a layer filter of `roadType is "highway"` would only be applied to roads index and not applied to the traffic volume measurements index.
```

Typo

Author: thank

> NOTE: Layer filters are not applied to the right side of *term joins*. You can apply a search request to the right side of *term joins* by setting the *where* clause in the join definition. For example, a layer with a term join where the left side is roads and the right side is traffic volume measurements, a layer filter of `roadType is "highway"` is only be applied to roads index and not applied to the traffic volume measurements index.

Reviewer 2: The last sentence is a little long.  How about breaking it into 2 sentences.

```suggestion
NOTE: Layer filters are not applied to the right side of *term joins*. You can apply a search request to the right side of *term joins* by setting the *where* clause in the join definition. For example, suppose you have a layer with a term join where the left side is roads and the right side is traffic volume measurements. A layer filter of `roadType is "highway"` is applied to the roads index, but not to the traffic volume measurements index.
```
